test/mocking_utils/stdio.hpp

```diff
@@ -52,7 +52,26 @@ using _vsnprintf_s_type =
 /// Patch _vscprintf with the given `replacement` in the given `scope`.
 #define patch__vscprintf(scope, replacement) patch(scope, _vscprintf, replacement)
 
-#endif  // defined(_WIN32)
+#else  // defined(_WIN32)
+
+// Deal with binary API nuances
+#ifndef NDEBUG
+
+/// Patch vsnprintf with the given `replacement` in the given `scope`.
+#define patch_vsnprintf(scope, replacement) patch(scope, vsnprintf, replacement)
+
+#else
+
+/// Patch vsnprintf with the given `replacement` in the given `scope`.
+#define patch_vsnprintf(scope, replacement) patch( \
+    scope, __vsnprintf_chk, [&](char * __str, size_t, int, size_t __size, \
```

Comment:
I think this kind of mocking is really fragile.

I prefer having less coverage and not doing this, but I guess it's not my call.

Reply:
> I think this kind of mocking is really fragile.

Indeed mocking system libraries this way is fragile, but as of today the alternative is not having coverage at all. Perhaps, eventually we can migrate towards fault injection (which has its own share of ifs and buts, like Release binary and Debug binaries not being the same).

> I prefer having less coverage

Generally speaking, I don't agree with reducing test coverage because it's inconvenient. I'd much rather look for a replacement if this becomes a pain point.